pkg/operator/provisioner.go

```diff
@@ -0,0 +1,207 @@
+	radosUser := fmt.Sprintf("%s-rbd-user", p.provConfig.clusterName)
+	secretRef := new(v1.LocalObjectReference)
+	secretRef.Name = fmt.Sprintf("%s-rbd-user", p.provConfig.clusterName)
+
+	pv := &v1.PersistentVolume{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: options.PVName,
+		},
+		Spec: v1.PersistentVolumeSpec{
+			PersistentVolumeReclaimPolicy: options.PersistentVolumeReclaimPolicy,
+			AccessModes:                   options.PVC.Spec.AccessModes,
+			Capacity: v1.ResourceList{
+				v1.ResourceName(v1.ResourceStorage): capacity,
+			},
+			PersistentVolumeSource: v1.PersistentVolumeSource{
+				RBD: &v1.RBDVolumeSource{
+					RBDImage:     imageName,
+					RBDPool:      p.provConfig.pool,
+					CephMonitors: monitors,
+					RadosUser:    radosUser,
+					SecretRef:    secretRef,
+					FSType:       "ext4",
+					ReadOnly:     false,
+				},
+			},
+		},
+	}
+	glog.Infof("successfully created Rook Block volume %+v", pv.Spec.PersistentVolumeSource.RBD)
+	return pv, nil
+}
+
+// createVolume creates a rook block volume.
+func createVolume(image, pool string, size int64, client rookclient.RookRestClient) (string, error) {
+	newImage := model.BlockImage{
+		Name:     image,
+		PoolName: pool,
+		Size:     uint64(size),
+	}
+
+	res, err := client.CreateBlockImage(newImage)
+	if err != nil {
+		return "", fmt.Errorf("Failed to create rook block image %s/%s: %v", pool, image, err)
+	}
+
+	return res, nil
+}
+
+// Delete removes the storage asset that was created by Provision represented
+// by the given PV.
+func (p *rookVolumeProvisioner) Delete(volume *v1.PersistentVolume) error {
+	rookClient, err := p.clusterManager.getRookClient(p.provConfig.clusterNamespace)
+	if err != nil {
+		return fmt.Errorf("Failed to get rook client: %v", err)
+	}
+
+	image := model.BlockImage{
+		Name:     volume.Spec.PersistentVolumeSource.RBD.RBDImage,
+		PoolName: p.provConfig.pool,
+	}
+
+	_, err = rookClient.DeleteBlockImage(image)
+	if err != nil {
+		return fmt.Errorf("Failed to delete rook block image %s: %v", volume.Name, err)
+	}
+	return nil
+}
+
+func parseClassParameters(params map[string]string) (*provisionerConfig, error) {
+	var cfg provisionerConfig
+	defaultCluster := "rook"
+
+	for k, v := range params {
+		switch strings.ToLower(k) {
+		case "pool":
+			cfg.pool = v
+		case "clusterNamespace":
+			cfg.clusterNamespace = v
+		case "clusterName":
+			cfg.clusterName = v
+		default:
+			return nil, fmt.Errorf("rookVolumeProvisioner: invalid option %q for volume plugin %s", k, "rookVolumeProvisioner")
+		}
+	}
+
+	if len(cfg.pool) == 0 {
+		return nil, fmt.Errorf("StorageClass for provisioner %s must contain 'pool' parameter", "rookVolumeProvisioner")
+	}
+
+	if len(cfg.clusterNamespace) == 0 {
+		cfg.clusterNamespace = defaultCluster
+	}
+
+	if len(cfg.clusterName) == 0 {
+		cfg.clusterName = defaultCluster
+	}
+
+	return &cfg, nil
+}
+
+func processMonAddresses(monAddresses []string) []string {
```

Comment:
Sounds good but since the provisioner.go is in the `operator` package, having a new file would force it to be in a different package. Im leaning to leaving this function here since it is only one function. We can refactor and move things around later once we see more utilities being used.